exercises/food-chain/test/Tests.hs

```diff
@@ -24,12 +25,16 @@ specs = describe "food-chain" $
             it "matches full song" $ song `shouldBe` lyrics
   where
 
-    lineAssertions = (go shouldBe `on` zip [1 :: Int ..] . lines) song lyrics
+    lineAssertions = zipWith checkLine [1 :: Int ..] $ zipMaybe (lines song) (lines lyrics)
```

Comment:
I was not very clever here. I suppose I can keep the `on` and extract the line number in `checkLine` (so `checkLine (lineno, got), (_, want)`)